### ISSUE TITLE
Context::current_time is not updated correctly

### ISSUE THREAD

**Issue by Jerboas86** [orottier/web-audio-api-rs#8]
Context::current_time always returns 0

Runs the following example to reproduce:

```rust
use web_audio_api::context::{AsBaseAudioContext, AudioContext};
use web_audio_api::node::{AudioNode, AudioScheduledSourceNode};

fn main() {
    let context = AudioContext::new();

    let osc1 = context.create_oscillator();
    osc1.connect(&context.destination());
    osc1.frequency().set_value_at_time(440., 0.);
    osc1.start();

    loop {
        println!("Time: {}", context.current_time());
    }
}
```

The example keeps printing 0:
```
Time: 0
Time: 0
Time: 0
Time: 0
```

from [web audio api spec](https://www.w3.org/TR/webaudio/#:~:text=Thus%2C%20for%20a%20running%20context%2C%20currentTime%20increases%20steadily%20as%20the%20system%20processes%20audio%20blocks%2C%20and%20always%20represents%20the%20time%20of%20the%20start%20of%20the%20next%20audio%20block%20to%20be%20processed.):

> Thus, for a running context, currentTime increases steadily as the system processes audio blocks, and always represents the time of the start of the next audio block to be processed.

**Comment by Jerboas86** [comment]
`AudioContext::current_time()` is derived from `BaseAudioContextInner.frames_played` field. But this field is not incremented at each render quantum.

```
struct BaseAudioContextInner {
    /// sample rate in Hertz
    sample_rate: SampleRate,
    /// number of speaker output channels
    channels: u32,
    /// incrementing id to assign to audio nodes
    node_id_inc: AtomicU64,
    /// message channel from control to render thread
    render_channel: Sender<ControlMessage>,
    /// number of frames played
    frames_played: AtomicU64,
    /// AudioListener fields
    listener_params: Option<AudioListenerParams>,
}
```

On the other hand, `RenderThread` struct got a frames_played field which is updated on each render quantum. But i don't think that this value can be accessed by `AudioContext`

```rust
pub(crate) struct RenderThread {
    graph: Graph,
    sample_rate: SampleRate,
    channels: usize,
    frames_played: AtomicU64,
    receiver: Receiver<ControlMessage>,
}
```

**Comment by orottier** [comment]
Yes, you are right. The `frames_played` should be wrapped in an `Arc` and should be identical to the `RenderThread`'s `frames_played`.

**Comment by Jerboas86** [comment]
I will try to make a PR tomorrow ;)

**Comment by Jerboas86** [comment]
I got a PR ready. I'll wait #10 , #7 , and #5 to be merged before pushing it

**Comment by orottier** [comment]
This was fixed in #18
